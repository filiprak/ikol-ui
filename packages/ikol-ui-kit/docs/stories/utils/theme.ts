import { useTheme } from '@/composables';

export function setupTheme(context: any) {
  const theme = useTheme();

  theme.is_dark.value = context?.globals?.theme === 'dark';

  return theme;
}
