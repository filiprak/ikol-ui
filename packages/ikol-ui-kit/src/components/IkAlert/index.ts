import IkAlert from './IkAlert.vue';
export type IkAlert = InstanceType<typeof IkAlert>;
export { IkAlert };
