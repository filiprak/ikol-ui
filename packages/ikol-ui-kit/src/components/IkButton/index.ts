import IkButton from './IkButton.vue';
export { IkButton };
