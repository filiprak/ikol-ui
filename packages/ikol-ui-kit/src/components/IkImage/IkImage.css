.ik-image {
    position: relative;
}

.ik-image--round {
    border-radius: 50%;
}

.ik-image--contain {
    object-fit: contain;
}

.ik-image--cover {
    object-fit: cover;
}

.ik-image--loading {
    position: relative;
    display: flex;
    justify-content: center;
    align-items: center;
}

.ik-image--loading .ik-image {
    position: absolute;
    width: 16px;
    height: 16px;
    opacity: 0;
}
