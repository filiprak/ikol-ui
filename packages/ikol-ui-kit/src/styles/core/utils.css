/* shadow */
.ik-shadow-0 {
    box-shadow: none;
}

.ik-shadow-1 {
    box-shadow: 0 1px 3px 0 var(--shadow-color-1);
}

.ik-shadow-2 {
    box-shadow: 0 2px 5px 0 var(--shadow-color-1);
}

.ik-shadow-3 {
    box-shadow: 0 2px 10px 0 var(--shadow-color-1);
}

.ik-shadow-4 {
    box-shadow: 0 2px 15px 0 var(--shadow-color-1);
}

.ik-shadow-5 {
    box-shadow: 0 2px 20px 0 var(--shadow-color-1);
}

.ik-shadow-6 {
    box-shadow: 0 3px 30px 0 var(--shadow-color-1);
}
