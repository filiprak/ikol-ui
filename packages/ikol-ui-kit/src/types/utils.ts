import type { App } from 'vue';

export interface VuePlugin {
    install: (app: App) => any
}

export enum UIDesignColor {
    primary = 'primary',
    success = 'success',
    error = 'error',
    accent = 'accent',
    default = 'default',
}

export enum UIElemSize {
    xs = 'xs',
    md = 'md',
    lg = 'lg',
}
