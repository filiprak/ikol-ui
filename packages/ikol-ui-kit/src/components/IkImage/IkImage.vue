<script lang="ts">
import '@/styles';
import './IkImage.css';
import type { Ref } from 'vue';
import { defineComponent, h, ref } from 'vue';
import { IkLoaderCircle } from '@/components/IkLoaderCircle';

export default defineComponent({
    name: 'IkImage',
    props: {
        src: {
            type: String,
            default: null,
        },
        alt: {
            type: String,
            default: null,
        },
        round: {
            type: Boolean,
            default: false,
        },
        size: {
            type: Number,
            default: null,
        },
        width: {
            type: Number,
            default: null,
        },
        height: {
            type: Number,
            default: null,
        },
        width_attr: {
            type: Number,
            default: null,
        },
        height_attr: {
            type: Number,
            default: null,
        },
        contain: {
            type: Boolean,
            default: false,
        },
        cover: {
            type: Boolean,
            default: false,
        },
        loader: {
            type: Boolean,
            default: false,
        },
    },
    setup(props, { slots }) {
        const classes = {
            'ik-image': true,
            'ik-image--empty': !props.src,
            'ik-image--round': props.round,
            'ik-image--contain': props.contain,
            'ik-image--cover': props.cover,
        };
        const loading = ref(props.loader);
        const error: Ref<Event | null> = ref(null);
        const width = props.width || props.size;
        const height = props.height || props.size;
        const styles: any = {};

        width && (styles.width = width + 'px');
        height && (styles.height = height + 'px');

        return () => {
            const img = h(props.src ? 'img' : 'div', {
                class: classes,
                src: props.src,
                loading: 'lazy',
                alt: props.alt || '{{_*en*Image_}}',
                width: props.width_attr || null,
                height: props.height_attr || null,
                style: styles,

                onLoad: props.loader
                    ? () => {
                        loading.value = false;
                        error.value = null;
                    }
                    : null,
                onError: props.loader
                    ? (event: Event) => {
                        loading.value = false;
                        error.value = event;
                    }
                    : null,

            }, slots.default?.());

            if (props.loader && loading.value) {
                return h('div', { style: styles, class: ['ik-image', 'ik-image--loading'] }, [
                    h(IkLoaderCircle, { indeterminate: true }),
                    img,
                ]);
            } else {
                return img;
            }
        };
    },
});
</script>
