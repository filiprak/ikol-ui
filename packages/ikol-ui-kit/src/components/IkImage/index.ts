import IkImage from './IkImage.vue';
export type IkImage = InstanceType<typeof IkImage>;
export { IkImage };
