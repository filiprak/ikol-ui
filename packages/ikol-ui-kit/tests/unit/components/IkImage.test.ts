import { renderShallow } from '@/tests/tools/utils';
import { IkImage } from '@/components/IkImage';

describe('IkImage', () => {
    const getInstance = (options = {}) => renderShallow(IkImage, options);

    it('renders with correct default state', () => {
        const wrapper = getInstance();
        expect(wrapper.classes()).toStrictEqual([
            'ik-image',
        ]);
        expect(wrapper.element.tagName).toBe('DIV');
    });

    it('renders with default slot', () => {
        const wrapper = getInstance({
            slots: {
                default: '<div class="test">Default slot test.</div>',
            },
        });
        expect(wrapper.text()).toContain('Default slot test.');
    });
});
