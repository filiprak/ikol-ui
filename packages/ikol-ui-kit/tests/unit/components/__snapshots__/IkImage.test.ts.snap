// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IkImage renders optimized for window size 1`] = `
<img
  alt="{{_*en*Image_}}"
  class="ik-image ik-image--contain"
  loading="lazy"
  src="http://image.com/image.png"
/>
`;

exports[`IkImage renders rounded 1`] = `
<img
  alt="{{_*en*Image_}}"
  class="ik-image ik-image--round"
  loading="lazy"
  src="http://image.com/image.png"
/>
`;

exports[`IkImage renders with correct default state 1`] = `
<div
  alt="{{_*en*Image_}}"
  class="ik-image ik-image--empty"
  loading="lazy"
/>
`;

exports[`IkImage renders with custom height 1`] = `
<img
  alt="{{_*en*Image_}}"
  class="ik-image"
  loading="lazy"
  src="http://image.com/image.png"
  style="height: 59px;"
/>
`;

exports[`IkImage renders with custom size 1`] = `
<img
  alt="{{_*en*Image_}}"
  class="ik-image"
  loading="lazy"
  src="http://image.com/image.png"
  style="width: 97px; height: 97px;"
/>
`;

exports[`IkImage renders with custom width 1`] = `
<img
  alt="{{_*en*Image_}}"
  class="ik-image"
  loading="lazy"
  src="http://image.com/image.png"
  style="width: 59px;"
/>
`;

exports[`IkImage renders with height attribute 1`] = `
<img
  alt="{{_*en*Image_}}"
  class="ik-image"
  height="59"
  loading="lazy"
  src="http://image.com/image.png"
/>
`;

exports[`IkImage renders with src and alt prop 1`] = `
<img
  alt="Image Alt"
  class="ik-image"
  loading="lazy"
  src="http://image.com/image.png"
/>
`;

exports[`IkImage renders with width attribute 1`] = `
<img
  alt="{{_*en*Image_}}"
  class="ik-image"
  loading="lazy"
  src="http://image.com/image.png"
  width="59"
/>
`;
